stable/python/_kubeless.py

```diff
@@ -97,18 +97,63 @@ def handler():
                 return res
 
 
+def preload():
+    """This is a no-op function used to start the forkserver."""
+    pass
+
+
 if __name__ == '__main__':
-    import sys
+    import logging
+    import multiprocessing as mp
     import requestlogger
+
+    mp_context = os.getenv('MP_CONTEXT', 'forkserver')
+
+    if mp_context == "fork":
+        raise ValueError(
+            '"fork" multiprocessing context is not supported because cherrypy is a '
+            'multithreaded server and safely forking a multithreaded process is '
+            'problematic'
+        )
+    if mp_context not in ["forkserver", "spawn"]:
+        raise ValueError(
+            f'"{mp_context}" is an invalid multiprocessing context. Possible values '
+            'are "forkserver" and "spawn"'
+        )
+
+    try:
+        ctx = mp.get_context(mp_context)
+
+        if ctx.get_start_method() == 'forkserver':
+            # Preload the current module and consequently also the user-defined module
+            # so that all the child processes forked from the forkserver in response to
+            # a request immediately have access to the global data in the user-defined
+            # module without having to load it for every request.
+            ctx.set_forkserver_preload([current_mod])
+
+            # Start the forkserver before we start accepting requests.
+            d = ctx.Process(target=preload)
+            d.start()
+            d.join()
+
+    except ValueError:
+        # Default to 'spawn' if 'forkserver' is unavailable.
+        ctx = mp.get_context('spawn')
+        logging.warn(
+            f'"{mp_context}" multiprocessing context is unavailable. Using "spawn"'
+        )
```

Comment:
what would be the use case in which is better to use the `spawn`ed process?

Reply:
None actually. The only reason I've included it is because not all platforms have support for `forkserver` yet.
Also, `spawn` is very safe and almost guaranteed to not have any problems and is available on all platforms which could be a quick workaround in case somebody faces an issue with `forkserver` (they just have to set an environment variable to switch to `spawn`).

`forkserver` is there for superior performance provided it is used properly. For example, If I were to use `spawn` with the example file I've included in the PR (`preload.py`), each and every request will take 20 seconds to complete because 20 seconds is how long it takes to load the two (hypothetical) ML models in that example (10 seconds to load one ML model). When I use `forkserver`, those models are only loaded once and that memory is shared with all the child processes, so each request only takes a few milliseconds to complete.

Reply:
sounds good